### ISSUE TITLE
Incorrect Line Breaks with Code Section

### ISSUE THREAD

**Issue by Merikanto** [jekyll/jekyll#7886]
So when I use the the code section in Jekyll (with Github Pages, Kramdown), sometimes I get incorrect line breaks. For example:

Normally a shell script should look like this:

```shell
for n in {0..4}; do
	if [[ $n == 2 ]]; then
		continue	
	fi
	echo $n
done

```

However, it looks like this in the renderer:

```shell
for n in {0..4}; do if [[ $n == 2 ]]; then continue  fi  echo 
$n done
```

This is my config for the page:

```
markdown: kramdown
highlighter: rouge

kramdown:
  input: GFM
  syntax_highlighter_opts:
    hard_wrap: true
    css_class: 'syntax'
```
I wonder which part went wrong! **I submit the issue under Kramdown's repo, but the people there said it's not really Kramdown's problem**. Thanks!

**Comment by ashmaroli** [comment]
I was not able to reproduce this.
I pasted this into Jekyll documentation's source-code and tested it. The result is:

![localhost_4000_tesst_](https://user-images.githubusercontent.com/12479464/68030780-4743a480-fce0-11e9-9a08-3d834a77135a.png)

**Comment by ashmaroli** [comment]
Please post the contents of your `Gemfile.lock` so that we can test with versions therein.
If there is no Gemfile in your site, please post the versions of Jekyll, kramdown and Rouge you have installed.

**Comment by Merikanto** [comment]
Rouge:  1.7.0

Kramdown: 1.4.2

Jekyll: 2.4.0

**Comment by ashmaroli** [comment]
Jekyll v2.4.0 is a really old version.
The latest version of Jekyll is: *`v4.0.0`*

However, it's best that you first [upgrade to `v3.8.6`](https://jekyllrb.com/docs/upgrading/2-to-3/), see if your issue gets resolved, (and then gradually [upgrade to `v4.0.0`](https://jekyllrb.com/docs/upgrading/3-to-4/) if necessary.)

**Comment by Merikanto** [comment]
Got it, I'll try that, thanks a lot!

(I'll close this for now then ~~)
